### PR TITLE
Run and record `adb shell screenrecord` during Android scenario app

### PR DESCRIPTION
**Pull request by matanlurey**
More grasping at straws to solve https://github.com/flutter/flutter/issues/145988.

At some point this could be moved to `--verbose` if it isn't useful.

This will automatically be copied into `${FLUTTER_LOGS_DIR}`, yay!

Example:

https://github.com/flutter/engine/assets/168174/c8caca7a-88ec-4d09-88bd-ebfc531f6512

### CODE REVIEW

**Comment by matanlurey** [comment]
Also /cc @camsim99 for funsies.

File: testing/scenario_app/bin/run_android_tests.dart [hunk 1]
```diff
@@ -361,14 +364,44 @@ Future<void> _run({
...
+
     await step('Running instrumented tests...', () async {
       final (int exitCode, StringBuffer out) = await pm.runAndCapture(<String>[
         adb.path,
         'shell',
         'am',
         'instrument',
         '-w',
-	'--no-window-animation',
+	      '--no-window-animation',
```
Comment (jonahwilliams) [inline testing/scenario_app/bin/run_android_tests.dart:404]: intentional?

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Review by gmackall — APPROVED** [review]
LGTM outside of weird spacing @jonahwilliams already pointed out

File: testing/scenario_app/bin/run_android_tests.dart (hunk 1 above)
Comment (matanlurey) [reply testing/scenario_app/bin/run_android_tests.dart]: Nope, that was weird!

**Pull request merged**